worlds/shivers/__init__.py

```diff
@@ -65,153 +71,184 @@ def create_regions(self) -> None:
-        librarylocation = self.random.choice([loc for loc in library_region.locations if not loc.name.startswith("Accessible:")])
+        library_location = self.random.choice(
+            [loc for loc in library_region.locations if not loc.name.startswith("Storage:")]
+        )
 
-        #Roll for which escape items will be placed in the Library
+        # Roll for which escape items will be placed in the Library
         library_random = self.random.randint(1, 3)
-        if library_random == 1: 
-            librarylocation.place_locked_item(self.create_item("Crawling"))
-
-            itempool = [item for item in itempool if item.name != "Crawling"]
-
-        elif library_random == 2: 
-            librarylocation.place_locked_item(self.create_item("Key for Library Room"))
-
-            itempool = [item for item in itempool if item.name != "Key for Library Room"]
-        elif library_random == 3: 
-            librarylocation.place_locked_item(self.create_item("Key for Three Floor Elevator"))
-            
-            librarylocationkeytwo = self.random.choice([loc for loc in library_region.locations if not loc.name.startswith("Accessible:") and loc != librarylocation])
-            librarylocationkeytwo.place_locked_item(self.create_item("Key for Egypt Room"))
-
-            itempool = [item for item in itempool if item.name not in ["Key for Three Floor Elevator", "Key for Egypt Room"]]
-
-        #If front door option is on, determine which set of keys will be used for lobby access and add front door key to item pool
-        lobby_access_keys = 1
+        if library_random == 1:
+            library_location.place_locked_item(self.create_item("Crawling"))
+            item_pool = [item for item in item_pool if item.name != "Crawling"]
+        elif library_random == 2:
+            library_location.place_locked_item(self.create_item("Key for Library"))
+            item_pool = [item for item in item_pool if item.name != "Key for Library"]
+        elif library_random == 3:
+            library_location.place_locked_item(self.create_item("Key for Three Floor Elevator"))
+            library_location_2 = self.random.choice(
+                [loc for loc in library_region.locations if
+                 not loc.name.startswith("Storage:") and loc != library_location]
+            )
+            library_location_2.place_locked_item(self.create_item("Key for Egypt Room"))
+            item_pool = [item for item in item_pool if
+                         item.name not in ["Key for Three Floor Elevator", "Key for Egypt Room"]]
+
+        # If front door option is on, determine which set of keys will
+        # be used for lobby access and add front door key to item pool
+        lobby_access_keys = 0
         if self.options.front_door_usable:
-            lobby_access_keys = self.random.randint(1, 2)
-            itempool += [self.create_item("Key for Front Door")]
+            lobby_access_keys = self.random.randint(0, 1)
```

Comment:
I'm confused why this only has a 50% chance of changing the value?

Reply:
Ah in the case that early or local lobby access is used and front door is also usable, then we give a 50/50 chance that either the player gets the original 3 key route or the front door.